### PR TITLE
Value restrictions and refactor

### PR DESCRIPTION
**Pull request by Afollet**
Added Value Restrictions to validate and refactored. Emptiness is now by default skipped
by the IllegalValueErrorCreator.

This includes both the written business logical for the metadata.xlsx validation and the
software that does the validation.

The code was heavily refactored and reorganized.

### CODE REVIEW

File: indexer/src/main/java/org/pdxfinder/dataloaders/updog/tablevalidation/rules/PdxValidationRuleset.java [hunk 1]
```diff
@@ -66,12 +73,12 @@ public TableSetSpecification generate(String provider) {
                     ColumnReference.of("metadata-model.tsv", "model_id"),
                     ColumnReference.of("metadata-sharing.tsv", "model_id")),
                 Relation.betweenTableColumns(
-                        Relation.validityType.one_to_many,
+                        validityType.ONE_TO_MANY,
```
Comment (mauroz77) [inline indexer/src/main/java/org/pdxfinder/dataloaders/updog/tablevalidation/rules/PdxValidationRuleset.java:75]: Could we capitalize the validityType enum?

Comment (Afollet) [reply indexer/src/main/java/org/pdxfinder/dataloaders/updog/tablevalidation/rules/PdxValidationRuleset.java:75]: That is odd.

Comment (Afollet) [reply indexer/src/main/java/org/pdxfinder/dataloaders/updog/tablevalidation/rules/PdxValidationRuleset.java:75]: Definitely.

**Comment by mauroz77** [comment]
PR looks great! I don't have any additional comments. I will approve so you can merge when travis passes.

**Review by mauroz77 — APPROVED** [review]

**Pull request merged**